File: frontend/src/types/notification.ts
export interface TriggerUser {
  id: string
  display_name: string
  avatar_url: string | null
}

export interface Notification {
  id: string
  action_type: string
  entity_type: string | null
  entity_id: string | null
  message: string
  is_read: boolean
  created_at: string
  trigger_user: TriggerUser | null
}

File: backend/app/schemas/notification.py
from pydantic import BaseModel


class TriggerUserResponse(BaseModel):
    id: str
    display_name: str
    avatar_url: str | None = None


class NotificationResponse(BaseModel):
    id: str
    action_type: str
    entity_type: str | None = None
    entity_id: str | None = None
    message: str
    is_read: bool
    created_at: str
    trigger_user: TriggerUserResponse | None = None


class NotificationListResponse(BaseModel):
    notifications: list[NotificationResponse]
    total: int
    unread_count: int


class BulkDeleteNotificationsRequest(BaseModel):
    notification_ids: list[str] | None = None
